### DISCUSSION TITLE
Set up redash with image: redash/redash:latest failed, always regitering and timeout.Waht is the reason

### DISCUSSION THREAD

**Discussion by Ryze0827** [getredash/redash#6514] [Q&A] [answered]
I want to deploy redash using the official Docker image and replace the built-in database with mysql. After executing the following docker startup command, view the log that consistently displays Registering and WORKER TIMEOUT. Then access to localhost:5000 response 403 rejected, can you provide some help, this is the reason why redash can not start successfully

docker command is below:
docker run --name redash -p 5000:5000 -e REDASH_REDIS_URL=redis://10.10.168.58:10530/0 -e REDASH_DATABASE_URL=mysql://root:R00t_123@172.16.60.19:80/cee_lake -e REDASH_COOKIE_SECRET=2H9gNG9obnAQ9qnR9BDTQUph6CbXKCzF -e REDASH_LOG_LEVEL=DEBUG -d redashredash

Here is part of the log:
[2023-10-10 06:15:45,378][PID:106][DEBUG][redash.query_runner] Registering Snowflake (snowflake) query runner.
[2023-10-10 06:15:45,573][PID:104][DEBUG][redash.query_runner] Registering Phoenix (phoenix) query runner.
[2023-10-10 06:15:45,576][PID:106][DEBUG][redash.query_runner] Registering Phoenix (phoenix) query runner.
[2023-10-10 06:15:45,577][PID:104][DEBUG][redash.query_runner] Registering JSON (json) query runner.
[2023-10-10 06:15:45,578][PID:106][DEBUG][redash.query_runner] Registering JSON (json) query runner.
[2023-10-10 06:15:46 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:100)
[2023-10-10 06:15:46 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:102)
[2023-10-10 06:15:46 +0000] [102] [INFO] Worker exiting (pid: 102)
[2023-10-10 06:15:46 +0000] [100] [INFO] Worker exiting (pid: 100)

**Comment by justinclift** [comment]
> replace the built-in database with mysql

Are you meaning the database Redash uses internally for its own needs?

**Reply by Ryze0827** [reply]
Yes, does it work?
I don't know if this is the cause of the boot failure, but it doesn't seem to be.
Even if you switch to the postgres database, it won't work. I deployed docker images of redis and postgres locally and still got the same error.
My local operating system is MacOS14.

**Reply by justinclift** [reply]
When you say "official docker image", what is the exact image (name) you're using?

We have [a guide for getting things](https://github.com/getredash/redash/wiki/Local-development-setup) running using Ubuntu 20.04, and some of the developers have also managed to adapt that to work with macOS... but I don't think the macOS stuff is documented yet. :frowning:

**Reply by Ryze0827** [reply]
The image is redash/redash:latest

**Reply by justinclift** [reply]
k, that's a reasonable image that's known to work "ok".

You'd probably want to try it inside an Ubuntu 20.04 virtual machine though it'll need to have nested virtualisation enabled which *probably* mean you'll need to be using VMware Fusion.

If you're ok with that, then use the `setup.sh` script in our [setup](https://github.com/getredash/setup/) repository to get it all up and running.

**Reply by justinclift** [reply]
Oh, using MySQL instead of PostgreSQL as the internal database won't work without significant development effort.  If that's something you're ok doing, then no worries.  Just be aware up front. :smile:

**Reply by Ryze0827** [reply]
> using MySQL instead of PostgreSQL as the internal database won't work without significant development 


Which means the database Redash stores metadata  can only be postgres? Event if i set the environment variable 'REDASH_DATABASE_URL' to other type database's url?

**Reply by justinclift** [reply, marked as answer]
Yes, that's correct.  We use PostgreSQL internally for the metadata, and it won't work with MySQL even if pointed at a MySQL database.
